src/net/client/cache.rs

```diff
@@ -0,0 +1,1177 @@
+//! A client cache.
+
+#![warn(missing_docs)]
+#![warn(clippy::missing_docs_in_private_items)]
+
+use crate::base::iana::{Class, Opcode, OptRcode, Rtype};
+use crate::base::name::ToDname;
+use crate::base::Dname;
+use crate::base::Message;
+use crate::base::MessageBuilder;
+use crate::base::ParsedDname;
+use crate::base::StaticCompressor;
+use crate::base::Ttl;
+use crate::dep::octseq::Octets;
+use crate::net::client::clock::{Clock, Elapsed, SystemClock};
+use crate::net::client::request::{
+    ComposeRequest, Error, GetResponse, SendRequest,
+};
+use crate::rdata::AllRecordData;
+use crate::utils::config::DefMinMax;
+use bytes::Bytes;
+use moka::future::Cache;
+use std::boxed::Box;
+use std::cmp::min;
+use std::fmt::{Debug, Formatter};
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::time::Duration;
+use std::vec::Vec;
+
+/// Configuration limit for the maximum number of entries in the cache.
+const MAX_CACHE_ENTRIES: DefMinMax<u64> =
+    DefMinMax::new(1_000, 1, 1_000_000_000);
```

Comment:
Given that the max of u64 is way higher than 1,000,000,000, where does this limit value come from?

Reply:
Just a random number that seems sensible. It might be too high even. Just have sane defaults to avoid accidents.

Reply:
Makes sense, but isn't clear from the code so if you could note that in a comment I would really appreciate that.

Reply:
I added a comment to set_max_cache_entries, which should show up in to documentation.